components/http/okHttp/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java

```diff
@@ -882,6 +882,16 @@ private void setBaseUrlForRequestInformation(@Nonnull final RequestInformation r
                     requestInfo.content == null
                             ? null
                             : new RequestBody() {
+                                @Override
+                                public long contentLength() {
+                                    if (requestInfo.contentLength != null) {
+                                        spanForAttributes.setAttribute(
+                                                "http.request_content_length",
+                                                requestInfo.contentLength);
+                                        return requestInfo.contentLength;
```

Comment:
I initially thought that it would make most sense to add it to the requestInformation via the suggestion to put in an overload, but seeing as it is a very commonly exposed method I agree that adding the check at the request adapter level and handling the exception there would be the least breaking method.